cmd/agent/app/check.go

```diff
@@ -112,7 +151,44 @@ var checkCmd = &cobra.Command{
 
 		allConfigs := common.AC.GetAllConfigs()
 
-		if breakPoint != "" {
+		if profileMemory {
+			// If no directory is specified, make a temporary one
+			if profileMemoryDir == "" {
+				profileMemoryDir, err = ioutil.TempDir("", "datadog-agent-memory-profiler")
+				if err != nil {
+					return err
+				}
+
+				defer func() {
+					_ = os.RemoveAll(profileMemoryDir)
+				}()
+			}
+
+			for idx := range allConfigs {
+				conf := &allConfigs[idx]
+				if conf.Name != checkName {
+					continue
+				}
+
+				var data map[string]interface{}
+
+				yaml.Unmarshal(conf.InitConfig, &data)
```

Comment:
The call to `yaml.Unmarshal` could return an err which should probably be tested

Reply:
Done!